src/features/bookOffer/helpers/bookingHelpers/bookingHelpers.ts

```diff
@@ -0,0 +1,124 @@
+import { OfferResponse, OfferStockResponse } from 'api/gen'
+import { BookingState, Step } from 'features/bookOffer/context/reducer'
+import { formatToFrenchDecimal } from 'libs/parsers'
+import { RadioSelectorType } from 'ui/components/radioSelector/RadioSelector'
+
+export function getButtonState(bookingState: BookingState) {
+  const { step, stockId, quantity, date, hour } = bookingState
+
+  switch (step) {
+    case Step.DATE: {
+      return date !== undefined
+    }
+    case Step.HOUR: {
+      return hour !== undefined || stockId !== undefined
+    }
+    case Step.PRICE: {
+      return stockId !== undefined
+    }
+    case Step.DUO: {
+      return quantity !== undefined
+    }
+    default: {
+      return false
+    }
+  }
```

Comment:
```suggestion
  switch (step) {
    case Step.DATE:
      return date !== undefined
    case Step.HOUR:
      return hour !== undefined || stockId !== undefined
    case Step.PRICE:
      return stockId !== undefined
    case Step.DUO:
      return quantity !== undefined
    default:
      return false
  }
```

inutile de mettre les parenthèses dans ton cas :)